docs/reinterpreting_effects.md

```diff
@@ -0,0 +1,359 @@
+# Reinterpreting effects
+
+This article assumes you are already familiar with [defining effects and their handlers](defining_effects.md).
+
+One of the nice aspects of effects is that they can support multiple effect handlers. Effects only specify actions, they don't actually perform them. Therefore, it's possible to "reinterpret" effects. There are multiple senses in which an effect can be reinterpreted:
+
+- Implementing an effect in terms of other effects. "Stacking" effects is a powerful tool for cleanly dividing implementations into the relevant abstraction layers with minimal leakage of implementation details.
+- Rewriting an effect and/or performing actions with the effect value and then performing the originally intended effect. This technique is conceptually similar to the middleware pattern commonly used in web applications.
+
+Let's explore both of these effect interpretation strategies with a small motivating example:
+
+✨ We would like to implement a client library for an [HTTP-based API that provides interesting cat facts](https://alexwohlbruck.github.io/cat-facts/docs/). ✨
+
+Let's break down some of the properties of the API client that would be desirable for a production use case:
+
+1. We would like to have our cat facts API be able to support different cat fact data sources in the future.
+2. We would like to be able to mock failure conditions (such as network connectivity issues) for testing purposes.
+3. We would like to be able to track timing metrics for how quickly we can retrieve cat facts.
```

Comment:
This is just a superb motivating example: at once cute, simple, and a lucid and relevant demonstration of real-world engineering problems.